UI: Makes 'Add [Support|Chat] Thread' floating

aethernet opened the pull request:
UI: Makes Add Thread button floating to save screen real estate.

Change-type: patch

Commit:
UI: Makes 'Add [Support|Chat] Thread' floating to save screen real estate

Change-type: patch

joshbwlng commented:
@aethernet Could you include some before/after screenshots in your PR description to make it clear what "floating" means? Thanks!

LucianBuzzo commented on apps/ui/lib/components/InfiniteList.tsx:
```diff
@@ -15,7 +15,9 @@ const ScrollArea = styled<any>(Box)`
 	overflow-y: auto;
 	display: flex;
 	flex-direction: column;
+	padding-bottom: 100px;
```
The additional bottom padding works fine here - its only an issue when you're at the most recent message

LucianBuzzo commented on apps/ui/lib/lens/list/Interleaved/component.tsx:
```diff
@@ -139,6 +139,20 @@ export const InterleavedList = (props: Props) => {
 			startReached={loadMoreContracts}
 			itemContent={itemContent}
 			overscan={10}
+			components={{
```
What is this Footer component for?

aethernet replied:
Adds a last "empty" element on the virtuoso list, which have the same purpose as the margin-bottom.

We cannot easily attach the margin-bottom to any of the internal virtuoso divs and doing on the wrapper reduces the scrollview.

Next step for this footer is to add a spinner while loading and a "end of scroll" message (or image)

LucianBuzzo commented on apps/ui/lib/lens/common/ViewFooter/ViewFooter.tsx:
```diff
@@ -14,7 +14,10 @@ import * as notifications from '../../../services/notifications';
 
 const Footer = styled(Flex)`
-	border-top: 1px solid #eee;
+	// border-top: 1px solid #eee;
```
You can remove this commented out code

aethernet commented:
@joshbwlng here's
<img width="485" alt="Cursor_and__52__3a563837-a2ad-4528-9559-5e4______Jellyfish" src="https://user-images.githubusercontent.com/447631/185178175-998a9671-7f36-4e85-ab75-823356b8aa4f.png">
<img width="505" alt="_52__3a563837-a2ad-4528-9559-5e4______Jellyfish" src="https://user-images.githubusercontent.com/447631/185178182-72334c9e-b350-4f7f-867f-f28c87e04afa.png">
<img width="588" alt="_52__6a3c84d9-7f5d-407f-a17b-917______Jellyfish" src="https://user-images.githubusercontent.com/447631/185178190-86394074-bb0b-4e55-8875-46bf26235cb0.png">

aethernet commented:
@joshbwlng before / after 

<img width="588" alt="_52__6a3c84d9-7f5d-407f-a17b-917______Jellyfish-2" src="https://user-images.githubusercontent.com/447631/185183423-c503ee1a-7ca0-462c-aa82-b3e97b8ad1a9.png">

<img width="505" alt="_52__3a563837-a2ad-4528-9559-5e4______Jellyfish-2" src="https://user-images.githubusercontent.com/447631/185183469-b6aa743e-2ca8-4f9b-98ad-970b72fd9807.png">

joshbwlng commented on apps/ui/lib/lens/list/Simple/component.tsx:
```diff
@@ -77,6 +77,20 @@ export default function SimpleList(props: Props) {
 			endReached={loadMoreContracts}
 			itemContent={itemContent}
 			overscan={5}
+			components={{
```
Looks like you have some duplicate code

joshbwlng commented:
@aethernet Would these changes not result in a button that would block visibility of text/information under it?

aethernet commented:
@joshbwlng yes indeed, but it only matter if you cannot scroll to move the test out of under the button.

That's the reason for the margin-bottom (on generic scroll lists) / empty footer (on virtuoso). It ensure you can always scroll the last line out of the button (and gives a little bit of comfort at the end as the last line of the feed is not glue to the side of the window/screen)

aethernet commented:
<img width="540" alt="_64__user-jmakivic___Jellyfish" src="https://user-images.githubusercontent.com/447631/185356407-f62cbb4e-a148-4837-83dc-5c929d274b0f.png">

LucianBuzzo committed:
UI: Use a rendition component for virtuoso footer.

This is a little cleaner, and uses styled-components instead of an
inline style so is slightly more performant.

Change-type: patch
Signed-off-by: Lucian Buzzo <lucian.buzzo@gmail.com>

LucianBuzzo commented on apps/ui/lib/lens/common/ViewFooter/ViewFooter.tsx:
```diff
@@ -14,7 +14,9 @@ import * as notifications from '../../../services/notifications';
 
 const Footer = styled(Flex)`
-	border-top: 1px solid #eee;
+	position: absolute;
```
This causes problems for lenses that don't infinite scroll, like the table lens. We may need to conditionally fix the button based on the lens being rendered 🤔

LucianBuzzo replied:
After some though I think the best approach will be to give controls to lenses to declaratively not render the footer, and adjust the other lenses to compensate for a floating button.

LucianBuzzo reviewed (commented):
I've pushed a commit that cleans up the Footer a little. Overall I like the change, however the flooring button causes layout issues for more static lenses, like the table and we need to resolve that issue before this can be merged.

LucianBuzzo committed:
UI: Adjust positioning of View footer to condense UI.

Change-type: patch
Signed-off-by: Lucian Buzzo <lucian.buzzo@gmail.com>

LucianBuzzo committed:
UI: Hide footer on chart lens

Adding a new contract from the footer doesn't make much sense on
a reporting lens like the chart lens.

Change-type: patch
Signed-off-by: Lucian Buzzo <lucian.buzzo@gmail.com>

LucianBuzzo committed:
UI: Add space for floating footer button in Table lens

Change-type: patch
Signed-off-by: Lucian Buzzo <lucian.buzzo@gmail.com>

LucianBuzzo committed:
UI: Make segment footer buttons float like the view footer buttons

Change-type: patch
Signed-off-by: Lucian Buzzo <lucian.buzzo@gmail.com>

LucianBuzzo committed:
UI: Reduce size of lens footer buttons

The button text has been replaced by icons and the text is now visible
on hover as a tooltip. Buttons now appear as small round controls.

Change-type: patch
Signed-off-by: Lucian Buzzo <lucian.buzzo@gmail.com>

LucianBuzzo commented:
@aethernet I tweaked this a bit more to make it play nicely with the other lenses, and added a similar affect for the segment footer (note to self: we need to unify those components). I also went a step further and switched to using an icon + tooltip for the buttons which makes them much smaller:

![image](https://user-images.githubusercontent.com/15064535/185379377-29c40450-48f6-4743-b001-3fe684e99d99.png)

aethernet commented:
@LucianBuzzo awesome !!
